adding-autodeployment/README.md

```diff
@@ -0,0 +1,100 @@
+
+## 2. Update `.travis.yml`
+If you are following from above, ignore the continuous integration instructions from above. We will be using a slightly different set of instructions from what is listed there.
+
+### before_deploy
+To utilize autopilot, you will need your travis deployment to have both cloud foundry and autopilot installed. To do this, you will modify your `before_deploy` section of your travis configuration file.
+
+First, look up on [cloud foundry's github](https://github.com/cloudfoundry/cli/releases) to find the latest release of the cf-cli for the **Linux 64 bit binary**.
+
+Add the following lines to the `before_deploy` section of your `.travis.yml` file, replacing `LATEST_VERSION` on the second line with the latest stable release you found above:
+```
+- export PATH=$HOME:$PATH
+- travis_retry curl -L -o $HOME/cf.tgz "https://cli.run.pivotal.io/stable?release=linux64-binary&version=LATEST_VERSION"
+- tar xzvf $HOME/cf.tgz -C $HOME
+- cf install-plugin autopilot -f -r CF-Community
+```
+
+### deploy
+In order to make use of conditional deployments to staging/master on different branches, add the following to the `deploy` section of the `.travis.yml`:
+```
+- provider: script
+  script: "./deploy.sh staging"
+  skip_cleanup: true
+  on:
+    branch: develop
+- provider: script
+  script: "./deploy.sh production"
+  skip_cleanup: true
+  on:
+    branch: master
+```
+For more on conditional deployments, you can check out the [cloud.gov documentation](https://cloud.gov/docs/apps/continuous-deployment/#using-conditional-deployments).
+
+## 3. Add `deploy.sh`
+To account for the branch-wise logic that is generated in the `.travis.yml`, we will be using a separate `deploy.sh` shell script that will be called by travis at the time of deployment.
+
+Follow the comments to adjust the script to suit your needs as you see fit. You will need to update the `spaces` to account for what is available on your particular cloud.gov.
+
+```
+set -e
+
+API="https://api.fr.cloud.gov" # This will be the following url if you are still deploying to E/W 'https://api.cloud.gov'
+ORG="gsa-acq-proto"            #Or whatever cloud.gov org you wish to place the app in
+APP_NAME=""                    # Enter your app name
+
+SPACE=$1
+
+if [ $# -ne 1 ]; then
+  echo "Usage: deploy <space>"
+  exit
+fi
+
+if [ $SPACE = 'production' ]; then
+  NAME=APP_NAME
+  MANIFEST="manifest.yml"
+elif [ $SPACE = 'staging' ]; then
+  NAME="$APP_NAME-staging"
+  MANIFEST="manifest-staging.yml"
+else
+  echo "Unknown space: $SPACE"
+  exit
+fi
+
+cf login -a $API -u $CF_USERNAME -p $CF_PASSWORD -o $ORG -s $SPACE
+cf zero-downtime-push $NAME -f $MANIFEST
+```
+
+## 4. Add manifests
```

Comment:
For a fully-automated deploy, it may be preferable to leave manifests out of version control (though a `manifest.yml.example` would be helpful) and build them on deploy using `cf create-app-manifest`. That would have the benefit of keeping services, instances, RAM, etc. consistent between the running app and its replacement.